aws/amazon-chime-sdk-js
# Leave DefaultAudioVideoController in a good state when the pre-started signaling channel closes unexpectedly.

**Description of changes:**

This can happen due to a simple one-minute timeout, or due to network
issues. If the channel was closed, the existing signaling task would
remain, and would have to fail again before a join could complete.

This commit registers a temporary observer, using it to clean up the
task if the connection is closed. Other code will create a fresh task
and proceed as usual.

**Testing**

Manual testing instructions:

* Join a meeting in one tab from the demo app.
* In another tab, open the demo app again.
* Check the box to pre-connect the signaling connection.
* Open the developer tools.
* Proceed to the device selection screen.
* Wait one minute. You'll see a message in the console about the websocket being closed.
* Finish joining the meeting and start your video. This should be instantaneous and work; both participants should see and hear each other almost immediately.

> 1. Have you successfully run `npm run build:release` locally?

In parts.

> 2. How did you test these changes?

See manual testing instructions above.

> 3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?

See above.

> 4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?

No.

> 5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?

No.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
